index.bs

```diff
@@ -1370,77 +896,778 @@ What configuration is the device in after it resets?
+        value of <code>0xJJMN</code> has major version <code>JJ</code>, minor version
+        <code>M</code> and subminor version <code>N</code>.
+
+    : <dfn>manufacturerName</dfn>
+    : <dfn>productName</dfn>
+    : <dfn>serialNumber</dfn>
+    ::
+        The {{USBDevice/manufacturerName}}, {{USBDevice/productName}} and
+        {{USBDevice/serialNumber}} attributes SHOULD contain the values of the <a>string
+        descriptors</a> indexed by the <code>iManufacturer</code>, <code>iProduct</code>
+        and <code>iSerialNumber</code> fields of the <a>device descriptor</a> if each is
+        defined.
+
+    : <dfn>configuration</dfn>
+    ::
+        The {{USBDevice/configuration}} attribute contains the currently selected
+        configuration for the device and SHALL be one of the {{USBConfiguration}} in
+        {{USBDevice/configurations}}.
+
+        The {{USBDevice/configuration}} getter steps are:
+          1. <a>For each</a> |configuration| of <a>this</a>.{{USBDevice/configurations}}:
+              1. If |configuration|.{{USBConfiguration/[[configurationValue]]}} is equal to 
+                  <a>this</a>.{{USBDevice/[[configurationValue]]}}, return |configuration|.
+          1. Return <code>null</code>.
+
+    : <dfn>configurations</dfn>
+    ::
+        The {{USBDevice/configurations}} attribute contains a <a>sequence</a> of
+        {{USBConfiguration}} representing configurations supported by the device.
+
+        The {{USBDevice/configurations}} getter steps are:
+          1. Return <a>this</a>.{{USBDevice/[[configurations]]}}.
+
+    : <dfn>opened</dfn>
+    ::
+        The {{USBDevice/opened}} attribute SHALL be set to <code>true</code> when the
+        device is opened by the current execution context and SHALL be set to
+        <code>false</code> otherwise.
+</dl>
+
+### Methods ### {#usbdevice-methods}
+
+<div algorithm="USBDevice.open()">
+    The <dfn for=USBDevice method>open()</dfn> method,
+    when invoked, MUST return a new {{Promise}}
+    |promise| and run the following steps <a>in parallel</a>:
+    1. Let |device| be <a>this</a>.
+    1. If |device| is no longer connected to the system, <a>reject</a> |promise|
+        with a {{NotFoundError}} and abort these steps.
+    1. If <code>|device|.{{USBDevice/opened}}</code> is <code>true</code>
+        <a>resolve</a> |promise| and abort these steps.
+    1. Perform the necessary platform-specific steps to begin a session with the
+        device. If these fail for any reason <a>reject</a> |promise| with a
+        {{NetworkError}} and abort these steps.
+    1. Set <code>|device|.{{USBDevice/opened}}</code> to <code>true</code> and
+        <a>resolve</a> |promise|.
+</div>
+
+<div algorithm="USBDevice.close()">
+    The <dfn for=USBDevice method>close()</dfn> method,
+    when invoked, MUST return a new {{Promise}}
+    |promise| and run the following steps <a>in parallel</a>:
+    1. Let |device| be the <a>this</a>.
```

Comment:
Use <a>this</a> directly anywhere you need the device instead of declaring a new variable the the same value. Like below, "If this.opened is false, resolve promise and abort these steps."

Reply:
Updated in https://github.com/WICG/webusb/pull/212/commits/e1b37d2a2893a75f3823dc2e2fe7635c22c1bfc4.